### PR TITLE
Canvas as image

### LINKED ISSUE #1596: Transform `Canvas` into an image
**Is your feature request related to a problem? Please describe.**
After drawing some things with a `Canvas` widget, one might want to transform these drawings into a bitmap.
This can help the user:
* Save the canvas drawings to a file
* Copy the drawings to clipboard
* Some other stuff

**Describe the solution you'd like**
There should be a `Canvas.as_bitmap()` method that converts the content of the `Canvas` into a map. The bitmap can be represented as:
* `toga.Image` instance
* Plain matrix (list of lists, not `numpy.array`!)
* a new `toga.Bitmap` class

**Describe alternatives you've considered**
An alternative would be to implement the objectives stated above directly, ie. implement method such as `Canvas.save_to_clipboard()` or `Canvas.save_to_path(path)`. In any case, `Canas.as_bitmap()` should be implemented, at least as an internal method.

### PR DESCRIPTION
**Pull request by saroad2**
Fixes #1596

Add API to convert `Canvas` content to binary data.

## PR Checklist:

- [X] All new features have been tested
- [X] All new features have been documented
- [X] I have read the **CONTRIBUTING.md** file
- [X] I will abide by the code of conduct

### CODE REVIEW

File: src/winforms/src/toga_winforms/widgets/canvas.py
Comment (freakboy3742) [inline src/winforms/src/toga_winforms/widgets/canvas.py:323]: The naming of this endpoint is a little misleading - it's returning image data, not an image.

We should either:
a. return an actual image from this API endpoint, or
b. rename this endpoint `get_image_data()`

I'm generally inclined to go with (b); the `get_` prefix would be consistent with the general theme of impl APIs being "get/set" driven (as they are all "interrogate a native layer to get/set the actual current state of the widget").

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
This generally looks good; I've flagged one fairly minor tweak about the naming of the API endpoint.

I'm also going to hold off merging this until after #1635 lands, as there's some overlap, and updating this PR to remove usage of `bind()` will be easier than finding `bind()` usage and up merging into #1635.

**Comment by mhsmith** [comment]
> I'm also going to hold off merging this until after #1635 lands, as there's some overlap, and updating this PR to remove usage of `bind()` will be easier than finding `bind()` usage and up merging into #1635.

#1635 has now been merged.

File: src/winforms/src/toga_winforms/widgets/canvas.py
Comment (saroad2) [reply src/winforms/src/toga_winforms/widgets/canvas.py:327]: How about `get_bitmap()`?

Because what is actually returning is a bitmap of the image.

Comment (freakboy3742) [reply src/winforms/src/toga_winforms/widgets/canvas.py:327]: "bitmap" isn't entirely accurate either - it's bit data, but not necessarily a bitmap; it could also be PNG/JPG/etc format encoded bit data.

**Review by freakboy3742 — APPROVED** [review]
I've made the changes to remove bind, and update the API for retrieving image data; I've also added a Cocoa implementation of image and canvas save, just to prove to myself that the APIs weren't Windows-specific.

**Review by freakboy3742 — APPROVED** [review]
Looks good; I've made the changes to remove bind, and update the API for retrieving image data; I've also added a Cocoa implementation of image and canvas save, just to prove to myself that the APIs weren't Windows-specific.

**Pull request merged**
